src/assets/CollectiblesController.ts

```diff
@@ -1148,6 +1153,105 @@ export class CollectiblesController extends BaseController<
 
     this.updateNestedCollectibleState(collectibles, ALL_COLLECTIBLES_STATE_KEY);
   }
+
+  /**
+   * Update colletible isTransacted status
+   *
+   * @param address - Hex address of the collectible contract.
+   * @param tokenId - Number that represents the id of the token.
+   * @param isTransacting - Collectible transacting status.
+   * @param transactionId - Hex id of the send transaction of the collectible.
+   */
+  updateCollectibleTransactionStatus(
+    address: string,
+    tokenId: string,
+    isTransacting: boolean,
+    transactionId: string,
+  ) {
+    const { allCollectibles } = this.state;
+    const { chainId, selectedAddress } = this.config;
+    const collectibles = allCollectibles[selectedAddress]?.[chainId] || [];
+    const index: number = collectibles.findIndex(
+      (collectible) =>
+        collectible.address === address && collectible.tokenId === tokenId,
+    );
+
+    if (index === -1) {
+      return;
+    }
+    const updatedCollectible: Collectible = {
+      ...collectibles[index],
+      isTransacting,
+      transactionId,
+    };
+
+    // Update Collectibles array
+    collectibles[index] = updatedCollectible;
+
+    this.updateNestedCollectibleState(collectibles, ALL_COLLECTIBLES_STATE_KEY);
+  }
+  /**
+   * Reset the collectible transaction id value and turn false the isTransacting prop
+   *
+   * @param transactionId - Hex id of the send transaction of the collectible.
+   * @returns
+   */
+  resetCollectibleTransactionStatusByTransactionId(
+    transactionId: string,
+  ): boolean {
+    const { allCollectibles } = this.state;
+    const { chainId, selectedAddress } = this.config;
+    const collectibles = allCollectibles[selectedAddress]?.[chainId] || [];
+    const index: number = collectibles.findIndex(
+      (collectible) => collectible.transactionId === transactionId,
+    );
+
+    if (index === -1) {
+      return false;
+    }
+    const updatedCollectible: Collectible = {
+      ...collectibles[index],
+      isTransacting: false,
+      transactionId: '',
+    };
+
+    // Update Collectibles array
+    collectibles[index] = updatedCollectible;
+
+    this.updateNestedCollectibleState(collectibles, ALL_COLLECTIBLES_STATE_KEY);
+    return true;
+  }
+
+  /**
+   * Returns if the token is being transacted
+   *
+   * @param address - Hex address of the collectible contract.
+   * @param tokenId - Number that represents the id of the token.
+   * @returns
+   */
+  getCollectibleTransactionStatus(
```

Comment:
I looked through CollectiblesController and there are a few places where we find a collectible by address and token id. What if we make this method more abstract so that those places can be cleaned up? In other words, perhaps we want to call this `findCollectibleByAddressAndTokenId`. Then we don't have to pull out only the properties we care about from the collectible, we simply return the collectible (if it was found) or undefined (if it was not).

Reply:
Should this be addressed in another PR @mcmire? Maybe we could raise another with this work and merge this one after that

Reply:
Right, I'm not talking about cleaning up the places that would use this, I just mean we add a more generic method for now, and then go back in another PR and clean up the places that could make use of the new method.

Reply:
Ok! I will refactor for a more generic method, thank you!